bittensor_cli/src/bittensor/chain_data.py

```diff
@@ -539,194 +577,34 @@ class SubnetInfo:
     burn: Balance
     owner_ss58: str
 
+    @classmethod
+    def _fix_decoded(cls, decoded: "SubnetInfo") -> "SubnetInfo":
+        d = decoded
+        return SubnetInfo(
+            netuid=d.netuid,
+            rho=d.rho,
+            kappa=d.kappa,
+            difficulty=d.difficulty,
+            immunity_period=d.immunity_period,
+            max_allowed_validators=d.max_allowed_validators,
+            min_allowed_weights=d.min_allowed_weights,
+            max_weight_limit=d.max_weights_limit,
+            scaling_law_power=d.scaling_law_power,
+            subnetwork_n=d.subnetwork_n,
+            max_n=d.max_allowed_uids,
+            blocks_since_epoch=d.blocks_since_last_step,
+            tempo=d.tempo,
+            modality=d.network_modality,
+            connection_requirements={
+                str(int(netuid)): u16_normalized_float(int(req))
+                for (netuid, req) in d.network_connect
+            },
+            emission_value=d.emission_values,
+            burn=Balance.from_rao(d.burn),
+            owner_ss58=decode_account_id(d.owner),
+        )
+
     @classmethod
     def list_from_vec_u8(cls, vec_u8: bytes) -> list["SubnetInfo"]:
         decoded = bt_decode.SubnetInfo.decode_vec_option(vec_u8)
-        result = []
-        for d in decoded:
-            result.append(
-                SubnetInfo(
-                    netuid=d.netuid,
-                    rho=d.rho,
-                    kappa=d.kappa,
-                    difficulty=d.difficulty,
-                    immunity_period=d.immunity_period,
-                    max_allowed_validators=d.max_allowed_validators,
-                    min_allowed_weights=d.min_allowed_weights,
-                    max_weight_limit=d.max_weights_limit,
-                    scaling_law_power=d.scaling_law_power,
-                    subnetwork_n=d.subnetwork_n,
-                    max_n=d.max_allowed_uids,
-                    blocks_since_epoch=d.blocks_since_last_step,
-                    tempo=d.tempo,
-                    modality=d.network_modality,
-                    connection_requirements={
-                        str(int(netuid)): u16_normalized_float(int(req))
-                        for (netuid, req) in d.network_connect
-                    },
-                    emission_value=d.emission_values,
-                    burn=Balance.from_rao(d.burn),
-                    owner_ss58=decode_account_id(d.owner),
-                )
-            )
-        return result
-
-
-custom_rpc_type_registry = {
```

Comment:
Does this need to be removed entirely?

Reply:
It would be a bit redundant as the same thing is now pulled from the chain